src/posit/connect/_api_call.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Protocol
+
+if TYPE_CHECKING:
+    from ._json import Jsonifiable
+    from .context import Context
+
+
+class ApiCallProtocol(Protocol):
+    _ctx: Context
+    _path: str
+
+    def _endpoint(self, extra_endpoint: str = "") -> str: ...
+    def _get_api(self, *, extra_endpoint: str = "") -> Jsonifiable: ...
+    def _delete_api(self, *, extra_endpoint: str = "") -> Jsonifiable | None: ...
+    def _patch_api(self, json: Jsonifiable | None, *, extra_endpoint: str = "") -> Jsonifiable: ...
+    def _put_api(self, json: Jsonifiable | None, *, extra_endpoint: str = "") -> Jsonifiable: ...
+
+
+def endpoint(ctx: Context, path: str, *, extra_endpoint: str = "") -> str:
+    return ctx.url + path + extra_endpoint
```

Comment:
pathinfo is the conventional name for this. You could also use `*args`. You should use posixpath.join to build the path to ensure all parts are separated by '/'

Something like...
```suggestion
def endpoint(ctx: Context, *path) -> str:
    return ctx.url + posixpath.join(*path)
```

Reply:
Perfect!